### ISSUE TITLE
[DISCUSSION][NEW FEATURE]acustic feedback of the remaining time of the combo

### ISSUE THREAD

**Issue by jefgenowitsch** [kroyee/SpeedBlocks#30]
What do you think of having the remaining combo time presented through audio?
Defenzor thought doing this with a synthesizer.

Here is an example:
https://www.youtube.com/watch?v=nHAe6nCY3rg

**Comment by kroyee** [comment]
He's just jumping on that poor thing the whole video. I didn't hear any synthesizer?
Wrong link?

**Comment by jefgenowitsch** [comment]
The first jumps: the sound gets higher.

BTW. I fell in love with speedblocks. Thank you for your effort!

**Comment by zDEFz** [comment]
my idea is to make use of a synthesizer
high time remaining = high pitch
low time remaining = low
what about it?
its about combo time remaining

**Comment by kroyee** [comment]
Ah, ok... Now I get it. :+1: 
SFML actually supports this, you can .pitch() a sound so it would be very easy to implements.
When would it play though? Everytime you add time to the timer, every second, every piece placed?

**Comment by zDEFz** [comment]
The Synthesizer should only play if a higher combo has been reached and if the remaining time gets to a certain degree to be able to estimate the remaining time without looking.

After i did a line clear i get a remember tone in the right pitch according to my combo / remaining time.

**Comment by Noir-** [comment]
Could be interesting. But this could be conflicting with the sound indicating the combo height. I think the easiest approach would be a warning tone every time a certain absolute amount of time (Like 2 sec, 1 sec, 0,5 sec) is left.

Btw: Can we change the title to something meaningful?

**Comment by zDEFz** [comment]
Or we implement something like this!

https://www.youtube.com/watch?v=2nJQfm-Raks

**Comment by kroyee** [comment]
I like that idea... a ticking clock that goes faster the less time you have (or pitch, or combined).
If it can be done fairly discrete so it's not a pain in the butt to have on.

**Comment by ghost** [comment]
@Noir- Something like: Combo and time based synthesizer.

**Comment by jefgenowitsch** [comment]
Just a first thought: (Think of a church bell, a tone for the hour and a certain tone for 15/30 min)
For speedblocks: Specific tone for each combo and for the remaining time something similar to this: https://www.youtube.com/watch?v=ueBlmU2Ilr0

But I also could imagine that this COULD be an overload of acoustic, when not perfectly designed.
However, I think that this could squeeze a bit more skill out of the players.

**Comment by ghost** [comment]
Just to add on to what people are saying about pitch based combo sounds, can you watch this and tell me if this is what you guys mean? 

[https://www.youtube.com/watch?v=Q-nmX5eQS2c](url)

**Comment by chinatsu** [comment]
@Expl0sive That's more an indication of how far along the combo has gone, the suggestion is to have a sound to indicate the remaining timer.

**Comment by ghost** [comment]
@jefgenowitsch "acoustic" is what you want to spell?

**Comment by zDEFz** [comment]
@Expl0sive your link is broken

**Comment by ghost** [comment]
@chinatsu Yes. correct , but this can also be used for changing the combo's pitch based on the remainder of the time for a specific combo (the one you are on) e.g. - 12 combo with half combo timer remaining. You clear at that exact time so it's 0.5 which plays the pitch assigned to the 12 combo at the 0.5 time. The same can be done for every other combo. So 0.125's 0.250's 0.50's etc...

**Comment by Noir-** [comment]
A clock ticking faster while time decreases sounds like a good idea. Everything else could be very noisy.

**Comment by kroyee** [comment]
I pushed a test of a ticking-clock implementation to branch tickingclock.

**Comment by kb1900** [comment]
Please see #32  @kroyee  which may not be easily compatible with a ticking clock set up

**Comment by zDEFz** [comment]
@kb1900 maybe we could might use that for ONLY 1/4 of time remaining OR in between.

**Comment by kroyee** [comment]
I merge master into tickingclock, so that branch is now up-to-date.
If anyone care to give it a try that would be great. You can experiment with changing the ticktock.wav to try different sounds. The sound loops so it can be how long/short as you like.
